test/resources/ostree.resource

```diff
@@ -3,86 +3,69 @@ Documentation       Keywords for OSTree-based systems
-    ${deploy_id}=    Get Booted Deployment Id
-    Remove Backups For Deployment    ${deploy_id}
-
-Rebase System And Verify
-    [Documentation]    Deploys given ostree ref on the host, reboots it,
-    ...    and checks if the host is healthy.
+    Make New SSH Connection
+
+    ${current_deploy}=    libostree.Get Booted Deployment Id
+    Should Be Equal As Strings    ${expected_deploy}    ${current_deploy}
+
+Deploy Commit Expecting A Rollback
+    [Documentation]    Deploys given ref and configures test agent for failing greenboot.
+    ...    It expects the system to roll back.
     [Arguments]    ${ref}
 
-    Rebase System    ${ref}
+    ${initial_deploy_id}=    Get Booted Deployment Id
+    ${deploy_id}=    Rebase System    ${ref}
+    TestAgent.Write
     Reboot MicroShift Host
 
-    Wait Until Greenboot Health Check Exited
-    System Should Be Healthy
+    Log To Console    "Failing ref deployed - waiting for system to roll back"
+    Wait Until Keyword Succeeds    20m    15s
+    ...    Current Deployment Should Be    ${initial_deploy_id}
+    Log To Console    "System rolled back - deploying target ref"
 
-    ${ref_after_reboot}=    Get Current Ref
-    Should Be Equal As Strings    ${ref_after_reboot}    ${ref}
+Deploy Commit Not Expecting A Rollback
+    [Documentation]    Deploys given ref and configures test agent for failing greenboot.
+    ...    It expects the system to roll back.
+    [Arguments]    ${target_ref}
 
-System Should Be Healthy
-    [Documentation]    Verifies if system is healthy by looking at health.json.
-    ...    If this runs after greenboot-healthcheck finishes, it verifies
-    ...    health of current boot.
+    ${initial_deploy_id}=    Get Booted Deployment Id
+    ${deploy_id}=    Rebase System    ${target_ref}
+    Reboot MicroShift Host
 
-    # Health of system with MicroShift 4.13 is implicitly checked by "Greenboot Health Check Exited"
-    ${version}=    MicroShift Version
-    IF    ${version.minor} == 13    RETURN
+    Log To Console    "Target ref deployed - starting health checking"
+    Wait Until Keyword Succeeds    10m    15s
+    ...    System Is Running Right Ref And Healthy    ${deploy_id}    ${initial_deploy_id}
 
-    ${health}=    Get Persisted System Health
-    Should Be Equal As Strings    ${health}    healthy
+System Is Running Right Ref And Healthy
+    [Documentation]    Checks if system is running right reference and is healthy
+    [Arguments]    ${expected_deploy}    ${initial_deploy}
 
-MicroShift 413 Should Not Have Upgrade Artifacts
-    [Documentation]    Verifies that host running MicroShift 4.13
-    ...    does not have upgrade related artifacts.
+    Make New SSH Connection
 
-    ${version}=    MicroShift Version
-    IF    ${version.minor} == 13
-        Path Should Not Exist    ${BACKUP_STORAGE}
-        Path Should Not Exist    /var/lib/microshift/version
+    ${current_deploy}=    libostree.Get Booted Deployment Id
```

Comment:
The logic here is looking at current and initial. Is it relevant what the "initial" deployment was, if that does not match the expected deployment? Is that check just for this fatal error handling?

Reply:
It's a shortcut to not wait whole 10/20 minutes just for it to fail because it rolled back

Reply:
That makes sense.